tests/packagedcode/test_rubygems.py

```diff
@@ -47,66 +47,39 @@
 # this is a multiple personality package (Java  and Ruby)
 # see also https://rubygems.org/downloads/jaro_winkler-1.5.1-java.gem
 
-# NOTE: this needs to be implemented first
-@expectedFailure
-class TestRubyGemspec(FileBasedTesting):
+# @expectedFailure
+class TestRubyGemspec(PackageTester):
     test_data_dir = os.path.join(os.path.dirname(__file__), 'data')
 
-    def check_gemspec(self, test_loc, expected_loc, regen=False):
-        test_loc = self.get_test_loc(test_loc)
-        expected_loc = self.get_test_loc(expected_loc)
-        results = rubygems.get_gemspec_data(test_loc)
-
-        try:
-            # fix absolute paths for testing
-            rel_path = results['loaded_from']
-            rel_path = [p for p in rel_path.split('/') if p]
-            rel_path = '/'.join(rel_path[-2:])
-            results['loaded_from'] = rel_path
-        except:
-            pass
-
-        if regen:
-            if py2:
-                mode = 'wb'
-            if py3:
-                mode = 'w'
-            with open(expected_loc, mode) as ex:
-                json.dump(results, ex, indent=2)
-        with io.open(expected_loc, encoding='UTF-8') as ex:
-            expected = json.load(ex)
-
-        assert sorted(expected.items()) == sorted(results.items())
-
     def test_rubygems_can_parse_gemspec_address_standardization_gemspec(self):
-        self.check_gemspec(
-            'rubygems/gemspec/address_standardization.gemspec',
-            'rubygems/gemspec/address_standardization.gemspec.expected.json')
-
-    def test_rubygems_can_parse_gemspec_arel_gemspec(self):
-        self.check_gemspec(
-            'rubygems/gemspec/arel.gemspec',
-            'rubygems/gemspec/arel.gemspec.expected.json')
+        test_file = self.get_test_loc('rubygems/gemspec/address_standardization.gemspec')
+        expected_loc = self.get_test_loc('rubygems/gemspec/address_standardization.gemspec.expected.json')
+        packages = rubygems.RubyGem.recognize(test_file)
+        self.check_packages(packages, expected_loc, regen=False)
 
-    def test_rubygems_modern_gemspec(self):
```

Comment:
Why removing that test too a few other below?

Reply:
These tests are added again :)